bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -119,7 +135,9 @@ bool PreProcessor::checkClientMsgCorrectness(ClientPreProcessReqMsgSharedPtr cli
 
 template <>
 void PreProcessor::onMessage<ClientPreProcessRequestMsg>(ClientPreProcessRequestMsg *msg) {
-  ClientPreProcessReqMsgSharedPtr clientPreProcessReqMsg(msg);
+  if (preProcessorMetrics_.requestReceived.Get().Get() % 10 == 0) metricsComponent_.UpdateAggregator();
```

Comment:
Note that this will update the aggregator only once in 10 received client messages.
If the messages rate is high, the aggregator will be updated very frequently and vice versa.
In the replica component, the aggregator is being updated on a timing basis.

Reply:
Well, if the aggregator is updated regularly, on a time basis, why do we need to do it explicitly here?

Reply:
> Well, if the aggregator is updated regularly, on a time basis, why do we need to do it explicitly here?

The aggregator is updated per concordMetrics::Component and PreProcessor has its own component instance.

Reply:
> Note that this will update the aggregator only once in 10 received client messages.
> If the messages rate is high, the aggregator will be updated very frequently and vice versa.
> In the replica component, the aggregator is being updated on a timing basis.

The target performance requirement is to support 950 operations per second. I want to update aggregator not more frequently than once in 100 ms.

Reply:
Thanks Yulia, I understand now.